01-indoor-floor-level-tracker/web-app/src/services/notehub/NotehubDataProvider.ts

```diff
@@ -226,9 +253,11 @@ export default class NotehubDataProvider implements DataProvider {
   }
 
   async getTrackerConfig(): Promise<TrackerConfig> {
-    const { notehubJsClient } = this;
-    const { api_key } = notehubJsClient.authentications;
-    api_key.apiKey = this.hubAuthToken;
+    await generateNotehubAuthToken(
+      this.notehubJsClient,
+      this.hubClientId,
+      this.hubClientSecret
+    );
```

Comment:
You're generating a new auth token on every call? I thought the idea was to generate the auth token once every 30 minutes and just renew it right before it's about to expire. Actually, feels like something the notehub-js library should do for you.

Reply:
For the sake of getting this out quickly as a sample for interested users to peruse, but since the OAuth approach is being reviewed again, I've asked the team if they can include some property on the token that will make it easier for me to determine when a token refresh is actually needed before a Notehub API call.

Reply:
The call that generates the token also returns an `expires_in`, which this app could store and use to conditionally generate tokens.

Or the library could do that. Still debating in my head whether having the library do this magically would be good or not.

Reply:
@paigen11 Ah sorry typed my comment before seeing yours. We do have this line in [the docs](https://dev.blues.io/reference/notehub-api/api-introduction/).

> Tokens automatically expire after 30 minutes, as reflected by the expires_in property (in seconds), which is returned alongside the access_token.

Is `expires_in` not actually there?

Reply:
It is. I'll take a closer look at it next week and clean this up some more.

Reply:
Refactored to store auth token data (including a custom generated `token_expiration_date` property) inside of cookies. Each time an API call is made anywhere in the app, the cookie's presence and validity are checked and if it's not set or expired a new auth token is generated, the API call is made, and the new auth data overwrites the previous cookie data (if there was one)